### PR TITLE
Fix result code calculation

### LINKED ISSUE #306: HMI sends response with WARNINGS resultCode in case requested vehicle data is available
**Bug Report**
HMI sends response with WARNINGS resultCode in case requested vehicle data is available

**Reproduction Steps**

   1. Start SDL, HMI, connect Mobile, register new App
   2. Send SubscribeVehicleData with speed = true
   3.  Send UnsubscribeVehicleData with speed = true

**Expected Behavior**

  1. HMI sends response to SDL:"{"jsonrpc":"2.0","id":58,"result":{"speed": {"dataType":"VEHICLEDATA_SPEED","resultCode":"SUCCESS"}, "code":0,"method":"VehicleInfo.SubscribeVehicleData"}}"
  2. HMI sends response to SDL: "{"jsonrpc":"2.0","id":60,"result":{"speed": {"dataType":"VEHICLEDATA_SPEED","resultCode":"SUCCESS"}
 ,"code":0,"method":"VehicleInfo.UnsubscribeVehicleData"}}"

**Observed Behavior**
1. HMI sends response to SDL:"{"jsonrpc":"2.0","id":58,"result":{"speed": {"dataType":"VEHICLEDATA_SPEED","resultCode":"SUCCESS"}
 , "code":21,"method":"VehicleInfo.SubscribeVehicleData"}}"
2. HMI sends response to SDL: "{"jsonrpc":"2.0","id":60,"result":{"speed": {"dataType":"VEHICLEDATA_SPEED","resultCode":"SUCCESS"}
,"code":21,"method":"VehicleInfo.UnsubscribeVehicleData"}}"

**OS & Version Information**
    sdl_hmi branch: develop
    commit: e5743af
    sdl_core branch: https://github.com/smartdevicelink/sdl_core/tree/6.1.0_RC

### PR DESCRIPTION
**Pull request by mkorniichuk**
Fixes #306 

This PR is **ready** for review.

### Summary
Current PR contains a fix for result code calculation function. Warnings result code was present if there were any successful subscriptions/unsubscriptions.

### CLA
- [x] I have signed [the CLA](https://docs.google.com/forms/d/e/1FAIpQLSdsgJY33VByaX482zHzi-xUm49JNnmuJOyAM6uegPQ2LXYVfA/viewform)

### CODE REVIEW

File: app/model/sdl/VehicleInfoModel.js [hunk 1]
```diff
@@ -406,7 +406,7 @@ SDL.SDLVehicleInfoModel = Em.Object.create(
         total_count: 0
       };
 
-      var code = SDL.SDLModel.data.resultCode.SUCCESS;
+      var code = SDL.SDLModel.data.resultCode.WARNINGS;
```
Comment (YarikMamykin) [inline app/model/sdl/VehicleInfoModel.js:409]: I think `code == SDL.SDLModel.data.resultCode.SUCCESS &&` on line 418 can be removed

Comment (mkorniichuk) [reply app/model/sdl/VehicleInfoModel.js]: @YarikMamykin good point! please, take a look 4a2e5b4cc73796b7002edec12fd85e0b5e33021c

File: app/model/sdl/VehicleInfoModel.js [hunk 2]
```diff
@@ -403,27 +403,37 @@ SDL.SDLVehicleInfoModel = Em.Object.create(
...
+        else if (subscriptions[key].resultCode == 'DATA_ALREADY_SUBSCRIBED' ||
+             subscriptions[key].resultCode == 'DATA_NOT_SUBSCRIBED') {
+          statistic.ignore++;
         }
       }
 
-      if (statistic.success_count > 0) {
-        return SDL.SDLModel.data.resultCode.WARNINGS;
+      var code = SDL.SDLModel.data.resultCode.WARNINGS;
+      if (statistic.total_count == 0) {
```
Comment (YarikMamykin) [inline app/model/sdl/VehicleInfoModel.js]: I think this one can be combined with `(statistic.ignore > 0)` on line 435

File: app/model/sdl/VehicleInfoModel.js [hunk 3]
```diff
@@ -403,27 +403,27 @@ SDL.SDLVehicleInfoModel = Em.Object.create(
...
       }
 
-      if (statistic.success_count > 0) {
-        return SDL.SDLModel.data.resultCode.WARNINGS;
+      var code = SDL.SDLModel.data.resultCode.WARNINGS;
+      if (statistic.total_count == 0 || statistic.ignore == statistic.total_count) {
+        code = SDL.SDLModel.data.resultCode.IGNORED;
+      }
+      else if (statistic.total_count == statistic.success_count) {
+        code = SDL.SDLModel.data.resultCode.SUCCESS;
```
Comment (AKalinich-Luxoft) [inline app/model/sdl/VehicleInfoModel.js:426]: @mkorniichuk as this is a result code, would be simpler to do:
```suggestion
    if (statistic.total_count == 0 || statistic.ignore == statistic.total_count) {
       return SDL.SDLModel.data.resultCode.IGNORED;
    }
    
    if (statistic.total_count == statistic.success_count) {
      return SDL.SDLModel.data.resultCode.SUCCESS;
    }
    
    return SDL.SDLModel.data.resultCode.WARNINGS;    
```

Comment (mkorniichuk) [reply app/model/sdl/VehicleInfoModel.js]: @AKalinich-Luxoft good point. Fixed in 061f5c11beaf6667108df20e261da1394b28a6b0

**Comment by IGapchuk** [comment]
@theresalech this PR is ready for review. Thank you

**Pull request merged**